src/command_modules/azure-cli-cognitiveservices/azure/cli/command_modules/cognitiveservices/_params.py

```diff
@@ -3,17 +3,53 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+import re
+import json
 from knack.arguments import CLIArgumentType
+from knack.log import get_logger
 
 from azure.cli.core.commands.parameters import (
     tags_type,
     resource_group_name_type,
     get_resource_name_completion_list)
 
+from azure.cli.core.commands.validators import validate_key_value_pairs
 
+logger = get_logger(__name__)
 name_arg_type = CLIArgumentType(options_list=['--name', '-n'], metavar='NAME')
 
 
+def validate_api_properties(string):
+    """ Extracts JSON format or 'a=b;c=d' format as api properties """
+    try:
+        return json.loads(string)
+    except ValueError:
+        result = validate_key_value_pairs(string)
+        if _is_suspected_json(string):
+            logger.warning('Api properties looks like a JSON format but not valid, interpreted as key=value pairs: %s',
+                           str(result))
+        return result
+
+
+def _is_suspected_json(string):
+    """ If the string looks like a JSON """
+    if string.startswith('{') or string.startswith('\'{') or string.startswith('\"{'):
+        return True
+    if string.startswith('[') or string.startswith('\'[') or string.startswith('\"['):
+        return True
+    if re.match(r"^['\"\s]*{.+}|\[.+\]['\"\s]*$", string):
+        return True
+
+    return False
```

Comment:
There is a function in core called "shell_safe_json_parse" that does a lot of this plus some other stuff. See if that will work for your purpose.  You basically attempt the JSON parse and if it fails, treat it as a string.

Reply:
Not exactly. 

Attempt the JSON parse
-> success, treat it as JSON
-> failed, call this method, then treat it as string:
     -> if this method say yes: give a warning.

Reason:
CLI can accept following formats:
```
{\"a\":\"b\"}
"{\"a\":\"b\"}"
```

But very likely the customer will try below formats:
```
{"a":"b"}
'{"a":"b"}'
'{\"a\":\"b\"}'
```

And I want to give warning is customer did that.

But `shell_safe_json_parse` should be useful to replace  json.loads, but I will have to catch CLIError

Reply:
Generally catching the CLIError is where you will proceed assuming it isn't JSON.

Reply:
This doesn't make it similar. We shouldn't validate this in the CLI. Different terminals have different approaches of escaping characters and the attempt to consolidate them in the CLI will complicate the text processing. We should simply threat the input as string and let `json` module to parse it and return whatever error message it signals.